src/components/EditPickersModal.vue

```diff
@@ -0,0 +1,244 @@
+    <ion-searchbar v-model="queryString" @keyup.enter="queryString = $event.target.value; findPickers()" />
+    <ion-row>
+      <ion-chip v-for="picker in selectedPickers" :key="picker.id">
+        <ion-label>{{ picker.name }}</ion-label>
+        <ion-icon :icon="closeCircle" @click="removePicker(picker.id)" />
+      </ion-chip>
+    </ion-row>
+
+    <ion-list>
+      <ion-list-header>{{ $t("Staff") }}</ion-list-header>
+      <div class="ion-padding" v-if="!pickers.length">
+        {{ 'No picker found' }}
+      </div>
+      <div v-else>
+        <ion-item v-for="(picker, index) in pickers" :key="index" @click="selectPicker(picker.id)">
+          <ion-label>
+            {{ picker.name }}
+            <p>{{ picker.externalId ? picker.externalId : picker.id }}</p>
+          </ion-label>
+          <ion-checkbox :checked="isPickerSelected(picker.id)"/>
+        </ion-item>
+      </div>
+    </ion-list>
+    <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+      <ion-fab-button :disabled="!selectedPickers.length" @click="confirmSave()">
+        <ion-icon :icon="saveOutline" />
+      </ion-fab-button>
+    </ion-fab>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import { 
+  IonButtons,
+  IonButton,
+  IonCheckbox,
+  IonChip,
+  IonContent,
+  IonHeader,
+  IonIcon,
+  IonFab,
+  IonFabButton,
+  IonTitle,
+  IonToolbar,
+  IonLabel,
+  IonItem,
+  IonList,
+  IonListHeader,
+  IonRow,
+  IonSearchbar,
+  modalController,
+alertController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, closeCircle, saveOutline } from "ionicons/icons";
+import { useStore } from "vuex";
+import { hasError, showToast } from '@/utils';
+import logger from "@/logger"
+import { UtilService } from "@/services/UtilService";
+import { translate } from "@/i18n";
+
+export default defineComponent({
+  name: "EditPickersModal",
+  components: { 
+    IonButtons,
+    IonButton,
+    IonCheckbox,
+    IonChip,
+    IonContent,
+    IonHeader,
+    IonIcon,
+    IonFab,
+    IonFabButton,
+    IonTitle,
+    IonToolbar,
+    IonLabel,
+    IonItem,
+    IonList,
+    IonListHeader,
+    IonRow,
+    IonSearchbar,
+  },
+  data () {
+    return {
+      selectedPickers: [] as any,
+      queryString: '',
+      pickers: [] as any,
+      editedPicklist: {} as any
+    }
+  },
+  mounted() {
+    this.findPickers()
+  },
+  props: ['selectedPicklist'],
+  methods: {
+    isPickerSelected(id: string) {
+      return this.selectedPickers.some((picker: any) => picker.id == id)
+    },
+    selectPicker(id: string) {
+      const picker = this.selectedPickers.some((picker: any) => picker.id == id)
+      if (picker) {
+        // if picker is already selected then removing that picker from the list on click
+        this.selectedPickers = this.selectedPickers.filter((picker: any) => picker.id != id)
+      } else {
+        this.selectedPickers.push(this.pickers.find((picker: any) => picker.id == id))
```

Comment:
We need to discuss how the system generated picklist will be handled.

Reply:
Implemented changes as discussed